PTGLgraphComputation/src/proteinstructure/Chain.java

```diff
@@ -53,6 +54,18 @@
     public ArrayList<Molecule> getMolecules() { return(molecules); }
     public ArrayList<String> getHomologues() { return(homologues); }
     
+    public Boolean isRnaChain() { return(this.getMoleculeType().contains("polyribonucleotide") ? true : false); }
+
+    public static Chain getChainByMacromolId(String inputId) {
+        ArrayList<Chain> allChains = FileParser.getChains();
+        for (Integer i = 0; i < allChains.size(); i++) {
+            Chain curChain = allChains.get(i);
+             if (curChain.getMacromolID().equals(inputId)) {
+                return(curChain);
+            }
+        }
+        return null;
+    }
```

Comment:
This is not the right place to put this function! Chain should only handle the specific chain object it was instantiated as. This function loops over all chains and this is why you need to get the chains from the Fileparser. So you could move the function there, but this is not directly related to the Fileparser either. Depending on from where you want to call it, you can move it to Main.